examples/oomkill.yaml

```diff
@@ -0,0 +1,47 @@
+programs:
+  # See:
+  # * https://github.com/iovisor/bcc/blob/master/tools/oomkill.py
+  # * https://github.com/iovisor/bcc/blob/master/tools/oomkill_example.txt
+  - name: oomkill
+    metrics:
+      counters:
+        - name: oom_kills
+          help: Count global and cgroup level OOMs
+          perf_map: events
+          labels:
+            - name: cgroup_path
+              size: 8
+              decoders:
+              - name: uint
+              - name: cgroup
+            - name: global_oom
+              size: 1
+              decoders:
+              - name: uint
+    kprobes:
+      oom_kill_process: count_ooms
+    code: |
+      #include <uapi/linux/ptrace.h>
+      #include <linux/oom.h>
+      #include <linux/memcontrol.h>
+
+      BPF_PERF_OUTPUT(events);
```

Comment:
Could you add a note why `BPF_PERF_OUTPUT` is preferred here over a plain `BPF_HASH`? Is it unbound cardinality of the data or something else?

Reply:
Precisely. There could be thousands of cgroup_ids across hundreds of nodes. I've added Prometheus counter flushing for this exact reason.